service/src/main/java/com/codeforcommunity/processor/ProtectedNeighborhoodsProcessorImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.codeforcommunity.processor;
+
+import com.codeforcommunity.api.IProtectedNeighborhoodsProcessor;
+import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.neighborhoods.SendEmailRequest;
+import com.codeforcommunity.dto.neighborhoods.UserEmailRecord;
+import com.codeforcommunity.requester.Emailer;
+import org.jooq.DSLContext;
+
+import java.util.List;
+
+import static org.jooq.generated.Tables.ADOPTED_SITES;
+import static org.jooq.generated.Tables.NEIGHBORHOODS;
+import static org.jooq.generated.Tables.SITES;
+import static org.jooq.generated.tables.Users.USERS;
+
+public class ProtectedNeighborhoodsProcessorImpl implements IProtectedNeighborhoodsProcessor {
+  private final DSLContext db;
+  private final Emailer emailer;
+
+  public ProtectedNeighborhoodsProcessorImpl(DSLContext db, Emailer emailer) {
+    this.db = db;
+    this.emailer = emailer;
+  }
+
+  @Override
+  public void sendEmail(JWTData userData, SendEmailRequest sendEmailRequest) {
+    List<Integer> neighborhoodIDs = sendEmailRequest.getNeighborhoodIDs();
+    String emailBody = sendEmailRequest.getEmailBody();
+
+    List<UserEmailRecord> userEmailRecords = db.select(USERS.EMAIL, SITES.ADDRESS)
+        .from(USERS)
+        .leftJoin(ADOPTED_SITES)
+        .on(USERS.ID.eq(ADOPTED_SITES.USER_ID))
+        .leftJoin(SITES)
+        .on(ADOPTED_SITES.SITE_ID.eq(SITES.ID))
+        .leftJoin(NEIGHBORHOODS)
+        .on(SITES.NEIGHBORHOOD_ID.eq(NEIGHBORHOODS.ID))
+        .where(NEIGHBORHOODS.ID.in(neighborhoodIDs)).fetchInto(UserEmailRecord.class);
```

Comment:
❓ Does this work if `neighborhoodIDs` is null?

Reply:
We actually changed it so that `neighborhoodIDs` can't be null and if someone wanted to send the email to everyone, they would pass in an empty list as `neighborhoodIDs`. We still have to actually implement this tho